NEWS.md

```diff
@@ -76,6 +76,8 @@ Language changes
   * Experimental feature: `x^n` for integer literals `n` (e.g. `x^3`
     or `x^-3`) is now lowered to `x^Val{n}`, to enable compile-time
     specialization for literal integer exponents ([#20530]).
+    `x^p` for `x::Number` and a literal `p=0,1,2,3` is now lowered to
```

Comment:
Yes, only these four exponents are inlined by this patch.  It's a minimal PR to match what inference.jl did before.   In the future, we may well want to inline more exponents.